src/sentry/static/sentry/app/components/forms/apiForm.tsx

```diff
@@ -40,14 +41,41 @@ export default class ApiForm extends Form<Props> {
     this.api.clear();
   }
 
+  getEnabledData() {
+    // Return a hash of data from non-disabled fields.
+
+    // Start with this.state.data and remove rather than starting from scratch
+    // and adding, because a) this.state.data is our source of truth, and b)
+    // we'd have to do more work to loop over the state.data Object and lookup
+    // against the props.children Array (looping over the Array and looking up
+    // in the Object is more natural). Maybe the consequent use of delete
+    // carries a slight performance hit. Why is yer form so big? 🤔
+
+    const data = {...this.state.data}; // Copy to avoid mutating state.data itself.
+    if (Array.isArray(this.props.children)) {
+      let child: any;
+      for (child of this.props.children) {
```

Comment:
So, just discovered that what we have been looking for was probably [`React.Children.forEach`](https://reactjs.org/docs/react-api.html#reactchildrenforeach). This should resolve some type issues we had here. (Sorry my React is rusty)

That said, since you've already resolved the type issues I'm not sure if we need to change it anymore.

Reply:
I like it! Implemented in b1739a1942. 👍